### DISCUSSION TITLE
Newbie Stuck- Not enough input arguments.  Error in AddExcitation

### DISCUSSION THREAD

**Discussion by 44xpre** [thliebig/openEMS-Project#223] [Q&A Usage]
Hello all:

I am very new to this, and just installed OpenEMS to prepare for some coursework in EM next semester, I've been trying to simulate simple antennas, but I am now stuck on something pretty (I think) basic that I just can't seem to figure out.

**_Not enough input arguments._**

_Error in AddExcitation (line 32)
CSX = AddProperty(CSX, 'Excitation', name, 'Type', type, 'Excite', excite, varargin{:});

Error in Opt42 (line 43)
CSX = AddExcitation(CSX, 'excitation', 0);_
 

Below is my code for the referenced lines.

(line 32) CSX = AddProperty(CSX, 'Excitation', name, 'Type', type, 'Excite', excite, varargin{:});

(line 43) CSX = AddExcitation(CSX, 'excitation', 0);
(line44)  CSX = AddBox(CSX, 'excitation', 0, [-SimBox(1)/2 -SimBox(2)/2 -SimBox(3)/2], [SimBox(1)/2 SimBox(2)/2 -SimBox(3)/2+lambda/4]);

Could anyone help me, to get pointed in the right direction? 

Thank you all, and sorry for the (most likely) stupid question
>>

**Comment by gadiLhv** [comment]
Hey @44xpre 

Assuming you are trying to add a port for an Antenna, the fastest way to go will be by using lumped ports.

[Here is an example](https://github.com/thliebig/openEMS/blob/1ccf0942477e9178b27f5e00dddd4d62bff78d29/matlab/examples/antennas/inverted_f.m#L123) from one of the openEMS matlab examples.

Adding an excitation is a bit more lower level. You get to choose between excitation types (E or H fields, for example) and other joys. 

For the newbie path, always start with existing code examples.
